### PR TITLE
fix: Add_enable_exactly_once_delivery_on_a_PubSub_subscription

### PR DESCRIPTION
**Pull request by hdavila146**
Hello, this 2 features have been remove from push subscriptions **_enable_message_ordering_** and **_enable_exactly_once_delivery_** they are only available now in pull susbcriptions only. @apeabody

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@hdavila146
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by apeabody** [comment]
Thanks for the contribution @hdavila146! - It would be best to have this covered by the Integration tests, could you enable `enable_exactly_once_delivery = true` for this example: https://github.com/terraform-google-modules/terraform-google-pubsub/blob/04ba61e291d6c548686ae8a8e81da1e09606f061/examples/simple/main.tf#L31

**Comment by comment-bot-dev** [comment]
@hdavila146
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
examples/simple/main.tf
--- old/examples/simple/main.tf
+++ new/examples/simple/main.tf
@@ -30,8 +30,8 @@
 
   pull_subscriptions = [
     {
-      name                 = "pull"
-      ack_deadline_seconds = 10
+      name                         = "pull"
+      ack_deadline_seconds         = 10
       enable_exactly_once_delivery = true
     },
   ]
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

**Comment by comment-bot-dev** [comment]
@hdavila146
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by hdavila146** [comment]
> Thanks for the contribution @hdavila146! - It would be best to have this covered by the Integration tests, could you enable `enable_exactly_once_delivery = true` for this example:
> 
> https://github.com/terraform-google-modules/terraform-google-pubsub/blob/04ba61e291d6c548686ae8a8e81da1e09606f061/examples/simple/main.tf#L31

@apeabody ready !!

File: main.tf [hunk 1]
```diff
@@ -197,10 +191,11 @@ resource "google_pubsub_subscription" "push_subscriptions" {
...
 
-  name    = each.value.name
-  topic   = var.create_topic ? google_pubsub_topic.topic.0.name : var.topic
-  project = var.project_id
-  labels  = var.subscription_labels
+  name                         = each.value.name
+  topic                        = var.create_topic ? google_pubsub_topic.topic.0.name : var.topic
+  project                      = var.project_id
+  labels                       = var.subscription_labels
+  enable_exactly_once_delivery = var.enable_exactly_once_delivery
```
Comment (apeabody) [inline main.tf:198]: I checked the CI example and the `true` wasn't passed through.  As enable_exactly_once_delivery is being passed in the `pull_subscriptions` map, you will probably need something like this:

```suggestion
  enable_exactly_once_delivery = lookup(
    each.value,
    "enable_exactly_once_delivery",
    false,
  )
```

**Comment by comment-bot-dev** [comment]
@hdavila146
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
main.tf
--- old/main.tf
+++ new/main.tf
@@ -191,10 +191,10 @@
 resource "google_pubsub_subscription" "pull_subscriptions" {
   for_each = var.create_subscriptions ? { for i in var.pull_subscriptions : i.name => i } : {}
 
-  name                         = each.value.name
-  topic                        = var.create_topic ? google_pubsub_topic.topic.0.name : var.topic
-  project                      = var.project_id
-  labels                       = var.subscription_labels
+  name    = each.value.name
+  topic   = var.create_topic ? google_pubsub_topic.topic.0.name : var.topic
+  project = var.project_id
+  labels  = var.subscription_labels
   enable_exactly_once_delivery = lookup(
     each.value,
     "enable_exactly_once_delivery",
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

**Comment by comment-bot-dev** [comment]
@hdavila146
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by hdavila146** [comment]
great @apeabody  I use your suggestion !!

**Comment by comment-bot-dev** [comment]
@hdavila146
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@hdavila146
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by apeabody — APPROVED** [review]
Thanks @hdavila146 - I added a validation test for the CI and everything looks good!

**Comment by EvgheniPopusoi** [comment]
Hm. Can you enable **enable_message_ordering** feature for push subscriptions? We use this in production for some projects.

**Comment by dkuji** [comment]
Hello, why was `enable_message_ordering` removed from push subscriptions?
I have been using this feature in my push subscriptions for at least 2 years now and it has never been removed from GCP in that time.

**Pull request merged**